application/datamanager/src/datamanager/main.py

```
@@ -1,123 +1,168 @@
+                    pl.from_epoch("t", time_unit="ms").dt.day().alias("day"),
+                ]
+            ).write_parquet(
+                bucket.daily_bars_path, partition_by=["year", "month", "day"]
+            )
+        except Exception as e:
+            logger.error(f"Error writing parquet file: {e}")
+            logger.error(traceback.format_exc())
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Failed to write data",
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)